server/src/main/java/io/unitycatalog/server/UnityCatalogServer.java

```diff
@@ -92,8 +93,32 @@ private void addServices(ServerBuilder sb) {
 
   public static void main(String[] args) {
     int port = 8080;
-    if (args.length > 0) {
-      port = Integer.parseInt(args[0]);
+    // Create Options object
+    Options options = new Options();
+
+    // Add port option
+    options.addOption("p", "port", true, "Port number to run the server on. Default is 8080.");
+    CommandLineParser parser = new DefaultParser();
+    try {
+      // Parse the command line arguments
+      CommandLine cmd = parser.parse(options, args);
+
+      // Check if port option is provided
+      if (cmd.hasOption("p")) {
+        port = Integer.parseInt(cmd.getOptionValue("p"));
```

Comment:
Let's avoid parsing this manually. We can define an [`Option.Builder#type(Class<?>)`](https://commons.apache.org/proper/commons-cli/apidocs/org/apache/commons/cli/Option.Builder.html#type(java.lang.Class)) and remove this parsing, along with the handling of `NumberFormatException`. Check [Converting (Parsing) Option Values](https://commons.apache.org/proper/commons-cli/usage.html) for more information.

Reply:
Thanks for pointing this out. Added the changes.